### PR TITLE
EDMFX entrainment/detrainment

### PR DESCRIPTION
**Pull request by trontrytel**
For now I'm adding entrainment ~ `1/z` and constant detrainment to simplify the testing. I'll keep the full formulation in the code, so that we don't get used to simple things too much.

Entrainment needs to know the buoyancy flux at the surface. Therefore the entrainment closure assumes that `get_surface_fluxes!` from `vertical_diffusion_boundary_layer_tendency` was called. Therefore we need to run EDMFX with:
`--vert_diff true` to switch it on, 
`--C_E 0` to zero out the coefficients
`--surface_scheme bulk` chose some surface scheme (for example bulk).

Closing #1417

### CODE REVIEW

File: src/tendencies/edmfx_closures.jl [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+                    thermo_params,
+                    ᶜts⁰[colidx],
+                    ᶜspecific⁰.e_tot[colidx],
+                ) -
+                ᶜentr_detrʲs.:($$j).detr[colidx] * TD.total_specific_enthalpy(
+                    thermo_params,
+                    ᶜtsʲs.:($$j)[colidx],
+                    ᶜspecificʲs.:($$j).e_tot[colidx],
+                ) -
+                ᶜp[colidx] / Y.c.ρ[colidx] * (
```
Comment (szy21) [inline src/tendencies/edmfx_closures.jl:155]: You shouldn't need this term here anymore:)

File: src/tendencies/edmfx_closures.jl [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            Y.c.sgsʲs.:($$j).ρa[colidx] * (
+                ᶜspecific⁰.q_tot[colidx] * ᶜentr_detrʲs.:($$j).entr[colidx] -
+                ᶜspecificʲs.:($$j).q_tot[colidx] *
+                ᶜentr_detrʲs.:($$j).detr[colidx]
+            )
+
+        # TODO - is this C3 correct?
+        @. Yₜ.f.sgsʲs.:($$j).w[colidx] += ᶠinterp(
+            ᶜentr_detrʲs.:($$j).entr[colidx] *
+            (C3(ᶜu⁰[colidx] - ᶜuʲs.:($$j)[colidx])),
```
Comment (szy21) [inline src/tendencies/edmfx_closures.jl:171]: Is `Y.f.sgsʲs.w` better than `ᶠinterp(C3(ᶜuʲs))` here?

**Review by szy21 — APPROVED** [review]
I didn't check the actual pi groups but I assume it's the same as the old code:) Otherwise it looks good to me, thanks!

File: src/tendencies/edmfx_closures.jl (hunk 1 above)
Comment (trontrytel) [reply src/tendencies/edmfx_closures.jl:155]: Great!

File: src/tendencies/vertical_diffusion_boundary_layer.jl [hunk 3]
```diff
@@ -263,6 +265,7 @@ function get_surface_fluxes!(Y, p, t, colidx, ::VerticalDiffusion)
         uₕ_int_phys,
         params,
         coupling,
+        buoy_flux_surface,
```
Comment (charleskawczynski) [inline src/tendencies/vertical_diffusion_boundary_layer.jl:268]: Shouldn't we just use `sfc_conditions[colidx].buoy_flux` in place of where this is used instead?

Comment (charleskawczynski) [reply src/tendencies/vertical_diffusion_boundary_layer.jl:268]: Or are they allowed to have different values for some reason?

Comment (trontrytel) [reply src/tendencies/vertical_diffusion_boundary_layer.jl:268]: I think you are right. For some reason I thought we are not caching the full `sfc_conditions`. Thanks. I'll open a PR with a fix.

Comment (trontrytel) [reply src/tendencies/vertical_diffusion_boundary_layer.jl:268]: And really thank you for reviewing!

**Pull request merged**